src/mixins/text_style.mixin.js

```diff
@@ -319,6 +319,86 @@
      */
     _deleteLineStyle: function(lineIndex) {
       delete this.styles[lineIndex];
+    },
+
+    /**
+     * Returns the array form of an object's styles property where styles are grouped in ranges
+     * rather than per character. This format is less verbose, and is better suited for storage
+     * so it is used in serialization (not during runtime).
+     * @return {Array}
+     */
+    _stylesToArray: function() {
+      // clone style structure to prevent mutation
+      var styles = fabric.util.object.clone(this.styles, true),
+          textLines = this.text.split('\n'),
```

Comment:
same here regarding computation
can you use `this._textLines` instead? and maybe pass this as an arg as well making these functions utils not relating to the instance?

Reply:
Using `this_textLines` would work for the i-text and text classes, but textbox lines have autowrapping so the `_textLines.lines` property treats autowrapped lines as their own lines (also these autowrapped line divisions aren't serialized, so they're not yet available during initialization until after initDimensions is called)

But yes, these methods should work fine as utils, I just thought it made sense to attach them to the instance since _stylesToArray uses `_hasStyleChangedForSvg` which is also attached to the instance.